From d7f5fb2c70187f04d4aedfeb52c737d18972f032 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 7 Jul 2021 15:43:57 +0200
Subject: [PATCH] ARM: dts: stm32: add USB OTG wakeup management on stm32mp131

USB OTG can be used to wakeup from stop mode on stm32mp131.
Add wakeup-source property, use EXTI #44 interrupt instead of GIC, and
attach USB OTG to "pd_core" power domain.

Change-Id: I2a1afd29c1b12aaf179eb44c3b333cb69f77bbac
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index bfe2f3fb087c..b54c8349ae05 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -828,13 +828,15 @@ usbotg_hs: usb@49000000 {
 			clock-names = "otg";
 			resets = <&rcc USBO_R>;
 			reset-names = "dwc2";
-			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti 44 IRQ_TYPE_LEVEL_HIGH>;
 			g-rx-fifo-size = <512>;
 			g-np-tx-fifo-size = <32>;
 			g-tx-fifo-size = <256 16 16 16 16 16 16 16>;
 			dr_mode = "otg";
 			otg-rev = <0x200>;
 			usb33d-supply = <&scmi_usb33>;
+			power-domains = <&pd_core>;
+			wakeup-source;
 			status = "disabled";
 		};
 
-- 
2.39.5

